test/unit/version-info.test.ts

```diff
@@ -0,0 +1,30 @@
+import { VersionInfo } from 'packages/next/src/client/components/react-dev-overlay/internal/container/Errors'
+import { parseVersionInfo } from '../../packages/next/src/server/dev/hot-reloader'
+
+describe('parse version info', () => {
+  test.each<
+    [
+      installed: string,
+      latest: string,
+      canary: string,
+      staleness: VersionInfo['staleness']
+    ]
+  >([
+    ['12.0.0', '13.0.0', '13.0.1-canary.0', 'stale-major'],
+    ['13.0.0', '13.1.0', '13.1.1-canary.0', 'stale-minor'],
+    ['13.1.1', '13.1.2', '13.1.3-canary.0', 'stale-patch'],
+    ['13.0.1-canary.0', '13.0.0', '13.0.1-canary.1', 'stale-prerelease'],
+    ['13.1.0', '13.1.0', '13.1.1-canary.0', 'fresh'],
```

Comment:
```suggestion
    ['13.1.0', '13.1.0', '13.1.1-canary.0', 'fresh'],
    ['13.1.1-canary.7', '13.1.0', '13.1.1-canary.7', 'fresh'],
```

Lets add a test for fresh canary